src/seedu/addressbook/commands/ClearCommand.java

```diff
@@ -10,10 +12,26 @@
             + "Example: " + COMMAND_WORD;
 
     public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_CONFIRMATION = "Are you sure you want to clear all contacts? Y/N";
+    public static final String MESSAGE_CANCELLED = "Cancelled, address book has not been cleared.";
+    public static final String MESSAGE_INVALID = "Invalid input, address book has not been cleared.";
 
     @Override
     public CommandResult execute() {
-        addressBook.clear();
-        return new CommandResult(MESSAGE_SUCCESS);
+        System.out.println(MESSAGE_CONFIRMATION);
+        Scanner scan = new Scanner(System.in);
+        String confirmClear = scan.nextLine();
+
+        if (confirmClear == "Y" || confirmClear == "y") {
+            addressBook.clear();
+            return new CommandResult(MESSAGE_SUCCESS);
+        } else {
+            if (confirmClear == "N" || confirmClear == "n") {
```

Comment:
Can simply use an else if instead of a nested if-else

Reply:
Could be better if just always convert confirmClear into lower case so that you will not have to deal with multiple cases.